vhost-user-backend/tests/vhost-user-server.rs

```diff
@@ -328,3 +354,38 @@ fn test_vhost_user_postcopy() {
     vhost_user_server(vhost_user_postcopy_listen);
     vhost_user_server(vhost_user_postcopy_end);
 }
+
+fn vhost_user_reset_device(path: &Path, barrier: Arc<Barrier>) {
+    let frontend = setup_frontend(path, barrier.clone());
+
+    // Signal that we are about to reset
+    barrier.wait();
+    // Wait until server has checked non-reset state
+    barrier.wait();
+
+    frontend.reset_device().unwrap();
+
+    // Signal reset is done
+    barrier.wait();
+}
+
+#[test]
+fn test_vhost_user_reset_device() {
+    vhost_user_server_with_fn(vhost_user_reset_device, |backend, barrier| {
+        // Wait until `vhost_user_reset_device()` is before reset
+        barrier.wait();
+        // Check non-reset state
+        assert!(backend.lock().unwrap().acked_features == 0xffff_ffff_ffff_ffff);
```

Comment:
Sounds like a good idea! How about adding an associated `const FEATURES: u64` to `MockVhostBackend`?

Reply:
Sounds good to me, maybe "REQUESTED_FEATURES" would make it slightly less generic... But I do not have any particular preference. I was only wondering about the meaning of the magic value (until I saw it being the same as above in `fn features()`).

Reply:
OK, I think I’ll make it `SUPPORTED_FEATURES`, then, because that value is what the mock “device” announces to be the features it can provide.

(`REQUESTED` sounds more like the ones the “guest driver” then actually requests to use, i.e. what `.acked_features` contains. It’s true that this value is really the one we should be verifying against, but that both depends on the device’s supported features and what the driver selects – I think it’s fair to rely on `setup_frontend()` to always request all of the provided feature bits.)